src/modules/aws_codedeploy/rpcs/start_deploy.ts

```diff
@@ -0,0 +1,218 @@
+import {
+  CodeDeploy,
+  CreateDeploymentCommandInput,
+  RevisionLocation,
+  waitUntilDeploymentSuccessful,
+} from '@aws-sdk/client-codedeploy';
+import { WaiterOptions } from '@aws-sdk/util-waiter';
+
+import { AwsCodedeployModule } from '..';
+import { AWS, crudBuilderFormat } from '../../../services/aws_macros';
+import { Context, RpcBase, RpcResponseObject } from '../../interfaces';
+
+/**
+ * Method for deploying a CodeDeploy application revision through a deployment group.
+ *
+ * Accepts the following parameters:
+ *
+ * - application: name of the application to deploy
+ *
+ * - deployment group: name of the deployment group to use
+ *
+ * - revision: complex type specifying the type and location of the revision to deploy
+ *
+ * - region: region where to trigger the deployment
+ *
+ * Returns following columns:
+ *
+ * - id: the ID of the triggered deployment
+ *
+ * - status: OK if the build was started successfully
+ *
+ * - message: Error message in case of failure
+ *
+ * @example
+ * ```sql TheButton[Deploy CodeDeploy application]="Deploy CodeDeploy application"
+ *   select * from start_deployment('test', 'test', '{
+ * "revisionType": "GitHub",
+ * "gitHubLocation": {
+ *   "repository": "iasql/iasql-codedeploy-example",
+ *   "commitId": "cf6aa63cbd2502a5d1064363c2af5c56cc2107cc"
+ * }
+ * }', 'us-east-2');
+ * ```
+ *
+ * @see https://github.com/iasql/iasql-engine/blob/main/test/modules/aws-codebuild-integration.ts#L313
+ * @see https://docs.aws.amazon.com/cli/latest/reference/deploy/create-deployment.html
+ *
+ */
+export class StartDeployRPC extends RpcBase {
+  /** @internal */
+  module: AwsCodedeployModule;
+
+  /** @internal */
+  outputTable = {
+    id: 'varchar',
+    status: 'varchar',
+    message: 'varchar',
+  } as const;
+
+  /** @internal */
+  startDeploy = crudBuilderFormat<CodeDeploy, 'createDeployment', string | undefined>(
+    'createDeployment',
+    input => input,
+    res => res?.deploymentId,
+  );
+
+  /** @internal */
+  call = async (
+    _dbId: string,
+    _dbUser: string,
+    ctx: Context,
+    applicationName: string,
+    deploymentGroupName: string,
+    revision: string,
+    region: string,
+  ): Promise<RpcResponseObject<typeof this.outputTable>[]> => {
+    if (!applicationName) {
+      return [
+        {
+          id: '',
+          status: 'KO',
+          message: 'Please provide the name of the CodeDeploy application to deploy',
+        },
+      ];
```

Comment:
I think you can refactor these into a `makeError` and `makeSuccess` private helper functions